### ISSUE TITLE
[QUERY] How to connect to Azurite blob store from sample blobs_client_example.c

### ISSUE THREAD

**Issue by andre2007** [Azure/azure-sdk-for-c#364]
**Query/Question**

It is unclear what to use to connect to a local Azurite blob store instance from sample `blobs_client_example.c`:

```C
#define TENANT_ID "tenant_id"
#define CLIENT_ID "devstoreaccount1"
#define CLIENT_SECRET "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw=="
#define URI "http://127.0.0.1:10000/"
```

- What to use for TENANT_ID?
- Is the URI correct or should it be http://127.0.0.1:10000/devstoreaccount1?

**Setup (please complete the following information if applicable):**
 - OS: Ubuntu (WSL)
 - IDE : VSC
 - Version : master
 
 **Information Checklist**
 Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Query Added
- [ ] Setup information Added

**Comment by vhvb1989** [comment]
@andre2007 , it might be an old version of the sample where the values were hardcoded.
From master: https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/storage/blobs/samples/src/blobs_client_example.c#L18

This is just a mapping to env variable names. 
So, we need to set env variables like, for instance:
```bash
export tenant_id=XXXXXXXX-XXX-XXXX-XXXX-XXXXXXXXXXXX
export client_id=XXXXXXXX-XXX-XXXX-XXXX-XXXXXXXXXXXX
export client_secret=xXxXxXxXxXxXxXxX...
#test_uri is the URL where click will send each operation request to (this is a keyvault example)
export test_uri=https://testingforc99.vault.azure.net
```

**Comment by andre2007** [comment]
I adapted the sample code to make the question more easy. Is it possible to connect with the sample code to a locally running blog storage (VSC Azurite extension) and what are the correct ENV variable values?

The values I posted are the public available values for a locally running blob store emulator.
https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azurite

**Comment by vhvb1989** [comment]
For the URI, according to the documentation, it should be http://127.0.0.1:10000/devstoreaccount1
see: https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azurite#endpoint-and-connection-url

The resource-path is added by SDK depending on the operation (upload, donwload, etc).

And about the other values (tenant, client and secret). This is the only currently supported authentication method for our SDK client to Azure.
Find more info about it here: https://docs.microsoft.com/en-us/azure/active-directory/azuread-dev/v1-oauth2-on-behalf-of-flow#service-to-service-access-token-request

Basically, this is a Service-to-service authentication where we need to register an app to get a client id and secret that sdk client will use to request a token.
So, you need to know if this authentication method is supported by Azurite. (their docs says they support all auth methods, so probably yes).

As a workaround, you could remove the authentication from sdk client at all (I can tell you how if you want to give it a try) just to work directly with Azurite.  Do you know how would it look an http request to Azurite using the public available values?

**Comment by jongio** [comment]
@andre2007 - I'm currently working with the Azurite team to add both HTTPS and Bearer Token support to Azurite.  In the meantime, you can use the stopgap that I blogged here: https://blog.jongallant.com/2020/02/azurite-https-defaultazurecredential/

**Comment by andre2007** [comment]
Thanks a  lot!
